storybook/webpack.config.js

```diff
@@ -2,18 +2,31 @@
  * External dependencies
  */
 const MiniCssExtractPlugin = require( 'mini-css-extract-plugin' );
+const path = require( 'path' );
 
 /**
  * Internal dependencies
  */
 const { getAlias, getMainConfig } = require( '../bin/webpack-helpers.js' );
+const tsConfig = require( '../tsconfig.json' );
+
+const aliases = Object.keys( tsConfig.compilerOptions.paths ).reduce(
+	( acc, key ) => {
+		const currentPath = tsConfig.compilerOptions.paths[ key ][ 0 ];
+		acc[ key.replace( '/*', '' ) ] = path.resolve(
+			__dirname,
+			'../' + currentPath.replace( '/*', '/' )
+		);
+		return acc;
+	},
+	{}
+);
```

Comment:
I switched to this configuration for getting aliases in the storybook webpack config because it's a bit more resilient than the other. I kept getting errors from using the base webpack config because it doesn't define all aliases (due to built bundles on externals). Whereas typescript does.  Typescript configuration for aliases is also pretty much the same as what storybook needs, so this aligns with that.